### ISSUE TITLE
k8s-extension authorization error

### ISSUE THREAD

**Issue by cmdbean** [Azure/azure-cli-extensions#5144]
### Related command
`az k8s-extension create`

### Extension name (the extension in question)
k8s-extension

### Description of issue (in as much detail as possible)

I was trying to install AzureML extensions to AKS using this URL.
https://docs.microsoft.com/en-us/azure/machine-learning/how-to-attach-kubernetes-anywhere?tabs=deploy-extension-with-cli%2Ccli

I tried to run the following command and the create command fails due to a permissions issue with an enterprise application I don't recognize.

```
az k8s-extension create --name test-extension --extension-type Microsoft.AzureML.Kubernetes --scope cluster --cluster-name test-cluster --resource-group <RESOURCE GROUP> --cluster-type managedClusters --configuration-settings enableInference=True allowInsecureConnections=True inferenceRouterServiceType=ClusterIP
```

This results in the following error
```
Error Message
Request failed to https://management.azure.com/subscriptions/<SUBSCRIPTION>/resourceGroups/<RESOURCE GROUP>/providers/Microsoft.ContainerService/managedclusters/<CLUSTER NAME>/extensionaddons/test-extention?api-version=2021-03-01. Error code: Forbidden. Reason: Forbidden.{"error":{"code":"AuthorizationFailed","message":"The client '<ENTERPRISE APP ID>' with object id '<ENTERPRISE APP ID>' does not have authorization to perform action 'Microsoft.ContainerService/managedclusters/extensionaddons/read' over scope '/subscriptions/<SUBSCRIPTION>/resourceGroups/<RESOURCE GROUP>/providers/Microsoft.ContainerService/managedclusters/test-cluster/extensionaddons/test-extention' or the scope is invalid. If access was recently granted, please refresh your credentials."}}
```
The following command succeeded
```
az k8s-extension list -g <RESOURCE GROUP> -c test-cluster --cluster-type managedClusters      
```
`-enable-aad --enable-azure-rbac`
The above options were specified when the cluster was created, and the following authorization was tried with no change in results

```
az role assignment create --role "Azure Kubernetes Service RBAC Admin" --assignee <ENTERPRISE APP ID> --scope /subscriptions/<SUBSCRIPTION>/resourceGroups/<RESOURCE GROUP>/providers/Microsoft.ContainerService/managedclusters/test-cluster

```

```
az version
{
  "azure-cli": "2.38.0",
  "azure-cli-core": "2.38.0",
  "azure-cli-telemetry": "1.0.6",
  "extensions": {
    "aks-preview": "0.5.91",
    "k8s-extension": "1.2.4",
    "ml": "2.6.1"
  }
}

```

-----

**Comment by yonzhan** [comment]
route to CXP team

**Comment by cmdbean** [comment]
I checked the error, this problem seems to be caused by the following request.

```
cli.azure.cli.core.sdk.policies: Request URL: 'https://management.azure.com/xxxxxxxxxx?api-Version=2022-03-01'
cli.azure.cli.core.sdk.policies: Request method: 'GET'
cli.azure.cli.core.sdk.policies: Request headers:
cli.azure.cli.core.sdk.policies:     'x-ms-client-request-id': 'xxxxxxxxxx'
cli.azure.cli.core.sdk.policies:     'CommandName': 'k8s-extension create'
cli.azure.cli.core.sdk.policies:     'ParameterSetName': '--name --extension-type --config --cluster-type --cluster-name --resource-group --scope --debug'
cli.azure.cli.core.sdk.policies:     'User-Agent': 'AZURECLI/2.38.0 (DEB) azsdk-python-azure-mgmt-kubernetesconfiguration/1.0.0 Python/3.10.5 (Linux-xxxxx)'
cli.azure.cli.core.sdk.policies:     'Authorization': '*****'
cli.azure.cli.core.sdk.policies: Request body:
cli.azure.cli.core.sdk.policies: This request has no body
urllib3.connectionpool: https://management.azure.com:443 "GET /subscriptions/xxxxxxx?api-Version=2022-03-01 HTTP/1.1" 200 None
cli.azure.cli.core.sdk.policies: Response status: 200
cli.azure.cli.core.sdk.policies: Response headers:
cli.azure.cli.core.sdk.policies:     'Cache-Control': 'no-cache'
cli.azure.cli.core.sdk.policies:     'Pragma': 'no-cache'
cli.azure.cli.core.sdk.policies:     'Transfer-Encoding': 'chunked'
cli.azure.cli.core.sdk.policies:     'Content-Type': 'application/json; charset=utf-8'
cli.azure.cli.core.sdk.policies:     'Content-Encoding': 'gzip'
cli.azure.cli.core.sdk.policies:     'Expires': '-1'
cli.azure.cli.core.sdk.policies:     'Vary': 'Accept-Encoding'
cli.azure.cli.core.sdk.policies:     'x-ms-ratelimit-remaining-subscription-reads': '11995'
cli.azure.cli.core.sdk.policies:     'Strict-Transport-Security': 'max-age=31536000; includeSubDomains'
cli.azure.cli.core.sdk.policies:     'api-supported-versions': '2019-11-01-Preview, 2021-05-01-preview, 2021-06-01-preview, 2021-09-01, 2021-11-01-preview, 2022-01-01-preview, 2022-03-01, 2022-04-02-preview'
cli.azure.cli.core.sdk.policies:     'X-Content-Type-Options': 'nosniff'
cli.azure.cli.core.sdk.policies:     'x-ms-request-id': 'xxxxxxxxxxxx'
cli.azure.cli.core.sdk.policies:     'x-ms-correlation-request-id': 'xxxxxxxxxxxx'
cli.azure.cli.core.sdk.policies:     'x-ms-routing-request-id': 'xxxxxxxxxxxx'
cli.azure.cli.core.sdk.policies:     'Date': 'Fri, 22 Jul 2022 12:48:23 GMT'
cli.azure.cli.core.sdk.policies: Response content:
```

**Comment by navba-MSFT** [comment]
@cmdbean Apologies for the late reply. Thanks for reaching out to us and reporting this issue. I was able to run the command just fine. Could you please update your `k8s-extension `extension and then run the below command to see if that helps ?

>az extension update -n k8s-extension

>az k8s-extension create --name test-extension --extension-type Microsoft.AzureML.Kubernetes --scope cluster --cluster-name MytestAKS --resource-group "My-Resources" --cluster-type managedClusters --configuration-settings enableInference=True allowInsecureConnections=True inferenceRouterServiceType=ClusterIP

**Comment by navba-MSFT** [comment]
@cmdbean I wanted to do quick follow-up to check if you had a chance to look at my above comment. Please let us know if you  are still seeing the same issue. Awaiting your reply.

**Comment by navba-MSFT** [comment]
@cmdbean The action plan is pending on you to update your `k8s-extension` extension as mentioned above and test the command again. Please feel free to reopen this thread, if that doesn't resolve this issue. We would be happy to help.

**Comment by danielvargasbm** [comment]
I am facing the same behaviour, I opened a new issue as I am facing this with cli v2: 
https://github.com/Azure/azure-cli/issues/25654

**Comment by dav1dli** [comment]
It looks like after some time the command which previously was producing an error is able to run. As if some token is refreshed. It is implied with the message: "If access was recently granted, please refresh your credentials.". How to trigger this refresh?
